libs/unity/library/Runtime/Scripts/Media/AudioReceiver.cs

```diff
@@ -158,6 +162,22 @@ protected void OnDestroy()
             AudioSettings.OnAudioConfigurationChanged -= OnAudioConfigurationChanged;
         }
 
+        protected new void Update()
+        {
+            base.Update();
+
+            // Check if _track has been changed by OnPaired/OnUnpaired and
+            // we need to start/stop streaming.
+            if (_track != null && !IsStreaming)
```

Comment:
This is essentially replacing an event-based implementation with a polling one, why?

Reply:
Note that this is just reapplying changes from #359.

The event-based implementation is not really so, as the handlers just enqueue operations on the Update thread. This doesn't work well if the state changes while the component is disabled - multiple actions are enqueued and then run all at once when the component is enabled again. It's difficult to keep the state consistent at each step while doing so. Putting the logic in Update() is straightforward.